oci/layout/oci_dest.go

```diff
@@ -299,6 +303,90 @@ func (d *ociImageDestination) CommitWithOptions(ctx context.Context, options pri
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// tryHardlinkLocalFile attempts to hardlink the specified file and digest it.
+func tryHardlinkLocalFile(dest *ociImageDestination, file string) (private.UploadedBlob, bool, error) {
+	blobFile, err := os.CreateTemp(dest.ref.dir, "oci-put-blob")
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	// Remove the file for os.Link() to work.
+	blobFile.Close()
+	os.Remove(blobFile.Name())
+	if err := os.Link(file, blobFile.Name()); err != nil {
```

Comment:
This is conceptually racy against the `CreateTemp` above. An alternative could be to create a temporary directory, and to create this link to a file in that temporary directory.